packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -2496,7 +2496,15 @@ class SemanticsOwner extends ChangeNotifier {
       visitedNodes.addAll(localDirtyNodes);
       for (SemanticsNode node in localDirtyNodes) {
         assert(node._dirty);
-        assert(node.parent == null || !node.parent.isPartOfNodeMerging || node.isMergedIntoParent);
+        if (node.parent != null) {
```

Comment:
Why take this condition out of the assert? The if only makes sense if asserts are enabled.

Reply:
I didn't think of that, I'll put it back into the assert